### ISSUE TITLE
BaseClass constructor doesn't make sense

### ISSUE THREAD

**Issue by adamduren** [ionic-team/ionic-native-google-maps#106]
I'm not sure I understand why the constructor is written as so.

https://github.com/ionic-team/ionic-native-google-maps/blob/c717aea6377de9662761f7f088826a23767d1e14/src/%40ionic-native/plugins/google-maps/index.ts#L1459-L1464

should it be this instead?:
```typescript
 constructor(_objectInstance?: any) { 
   if (!_objectInstance) { 
     _objectInstance = new (GoogleMaps.getPlugin().BaseClass)(); 
   } 
   this._objectInstance = _objectInstance; 
 } 
```

**Comment by adamduren** [comment]
The `this.` in the if branch seems unintended, unless I'm missing something.

**Comment by adamduren** [comment]
It could also be written as:
```typescript
 constructor(_objectInstance?: any) { 
   this._objectInstance = _objectInstance ||  new (GoogleMaps.getPlugin().BaseClass)();
 } 
```

or even:
```typescript
  constructor(_objectInstance: any = new (GoogleMaps.getPlugin().BaseClass)()) {}
```

**Comment by wf9a5m75** [comment]
I prefer the person who send a PR instead of blaming if you find a mistake.

**Comment by wf9a5m75** [comment]
Fixed. And released as `@ionic-native/google-maps@5.0.0-beta.25`.
Thank you for pointing out.

**Comment by adamduren** [comment]
Thank you. I'm working on a method to be able to test the ionic-native version of the plugin. I've got a prototype working. I still need to document how it works but will be presenting something in the next couple of days.
